### DISCUSSION TITLE
Cannot select the element with wdio

### DISCUSSION THREAD

**Discussion by HoLingVN** [webdriverio/webdriverio#9769] [Community]
I want to select the 1st button in this div for my test:

```
<div id="group_button" class="text-center background-color-ecodrop common-padding-bottom-10">
                <button ng-click="filterPupNotCollect()" class="custom-button-map enable-button-map" ng-disabled="listPUPMap.length == 0" ng-class="{'enable-button-map': isShowFormCollect == false, 'disable-custom-button-map': listPUPMap.length == 0}">Déposer en<br>déchèterie</button>
                <button id="button_filter_pup_collect" ng-click="filterPupCollect()" class="custom-button-map common-margin-left-10" ng-class="{'enable-button-map': isShowFormCollect, 'disable-custom-button-map': listCollect.length == 0}" ng-blur="hideTooltip()">Commander une<br>collecte</button>
            </div>
```
I tried many ways as the instruction of wdio: https://webdriver.io/docs/selectors/#chain-selectors But they all didn't work They return with the error `elem[prop] is not a function` These are what I tried:

```
return $('.text-center.background-color-ecodrop.common-padding-bottom-10 button:nth-child(1)')
//return $$('#group_button')[0]
//return $('.text-center.background-color-ecodrop.common-padding-bottom-10 .custom-button-map.enable-button-map:nth-child(1)')
 //return $('#group_button').$('button*=Déposer en')
 //return $('//*[@id="group_button"]/button[1]')
```
Any suggestion how to solve this? Thanks in advance

**Comment by christian-bromann** [comment]
How about:

```ts
const firstButton = await $$('#group_button')[0]
```

> They return with the error `elem[prop] is not a function` These are what I tried:

Can you provide a reproducible example, e.g. a html page that can be used to reproduce this?

**Reply by HoLingVN** [reply]
Thanks but it still doesn't work. 
I was testing this page: https://ecodrop.apac.positivethinking.tech/uat/ecodrop/#/app/craftsmen/deposite/create

And here my page objects:
[test.zip](https://github.com/webdriverio/webdriverio/files/10778946/test.zip)

I tried many ways as the wdio document but it still doesn't recognize the element

**Reply by christian-bromann** [reply]
Make sure to wait until the element is available, e.g. via `await $('#group_button button').waitForExist()`
